src/deprecated.jl

```diff
@@ -97,9 +97,21 @@ function _upgrade_env(pkgs::Dict)
         mktempdir() do tmp
             Pkg.activate(tmp)
 
-            # We construct an array of PackageSpecs to avoid ordering problems with adding
-            # each package individually
-            Pkg.add([Pkg.PackageSpec(; name=key, version=value) for (key, value) in pkgs])
+            # We construct an array of PackageSpecs to avoid ordering problems with
+            # adding each package individually
+            try
+                Pkg.add([
+                    Pkg.PackageSpec(; name=key, version=value) for (key, value) in pkgs
+                ])
+            catch e
+                # Warn about failure and fallback to simply trying to install the pacakges
+                # without version constraints.
+                warn(LOGGER) do
+                    "Failed to construct an environment with the provide package version " *
+                    "($pkgs): $e.\n Falling back to simply adding the packages."
+                end
+                Pkg.add([Pkg.PackageSpec(; name=key) for (key, value) in pkgs])
```

Comment:
What behaviour does this have for packages with the same name (distinguishable by UUID)?

Reply:
This PR won't change the behaviour, but `Pkg.add` will prompt the user to select the registry where the package lives.

Reply:
I guess because this is for handling old files, and thus we don't have the UUID recorded, that is the best we can do

Reply:
That's correct, there's only so much we can do when upgrading from older JLSO format versions.